airflow-core/src/airflow/dag_processing/processor.py

```diff
@@ -206,7 +206,6 @@ class DagFileParsingResult(BaseModel):
     type: Literal["DagFileParsingResult"] = "DagFileParsingResult"
 
 
-@functools.cache
```

Comment:
Why remove this?

Reply:
Because it's already cached here:

```
    @functools.cached_property
    def client(self) -> Client:
```

Reply:
Yes the client is already cached and that's what is actually used. I don't think we need to cache the `in_process_api_server`. Let me know if you want me to add it back.